packages/customer-effort-score/src/customer-feedback-modal/index.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import PropTypes from 'prop-types';
+import {
+	Button,
+	Modal,
+	RadioControl,
+	__experimentalText as Text,
+	TextareaControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import classnames from 'classnames';
+
+/**
+ * Provides a modal requesting customer feedback.
+ *
+ * A label is displayed in the modal asking the customer to score the
+ * difficulty completing a task. A group of radio buttons, styled with
+ * emoji facial expressions, are used to provide a score between 1 and 5.
+ *
+ * A low score triggers a comments field to appear.
+ *
+ * Upon completion, the score and comments is sent to a callback function.
+ *
+ * @param {Object} props                       Component props.
+ * @param {Function} props.recordScoreCallback Function to call when the results are sent.
+ * @param {string} props.label                 Question to ask the customer.
+ */
+function CustomerFeedbackModal( { recordScoreCallback, label } ) {
+	const options = [
+		{
+			label: __( 'Very difficult', 'woocommerce-admin' ),
+			value: '1',
+		},
+		{
+			label: __( 'Somewhat difficult', 'woocommerce-admin' ),
+			value: '2',
+		},
+		{
+			label: __( 'Neutral', 'woocommerce-admin' ),
+			value: '3',
+		},
+		{
+			label: __( 'Somewhat easy', 'woocommerce-admin' ),
+			value: '4',
+		},
+		{
+			label: __( 'Very easy', 'woocommerce-admin' ),
+			value: '5',
+		},
+	];
+
+	const [ score, setScore ] = useState( NaN );
```

Comment:
It doesn't seem right to initialize with `NaN`. I think `NaN` implies there is some sort of error, which isn't the case here.

I think it was better to keep the internal score state as a string, since that is how the radio control natively works and avoids type juggling. I think it is good to convert it to an integer for the `sendScore` parameter though.

Reply:
I don't have a strong opinion either way. I'm happy to roll with this and we can revisit it later if it still feels janky.

Reply:
Yeah, the use of `NaN` feels a bit too 'clever', but I liked it because it allowed me to not have to specifically account for `null` or `undefined` elsewhere (since `NaN.toString()` works).

In general, I like to keep data in its 'natural' form, and only convert it when needed to fit UI/external integration needs. Score is, at heart, a number. So, I like the idea of treating it as a number whenever possible in our business logic.

But, I don't feel super strongly about this particular case.